### PR TITLE
fix(chainselection): implausible tip check

### PR DESCRIPTION
**Pull request by wolf31o2**
---
## Summary by cubic
Fix implausible tip validation in `chainselection` to compare new tips against a peer-based reference and use overflow-safe boundary math. This blocks spoofed high tips without rejecting legitimate peers during bootstrap and sync.

- **Bug Fixes**
  - `UpdatePeerTip` now checks known peers against their previous tip (even if 0), new peers against the best known peer tip, and accepts the first peer when no reference exists.
  - Added `safeAddUint64` and used it for k math in tip validation and in the "implausibly-behind" peer filter to avoid overflow near `MaxUint64`.
  - Tests expanded for initial sync with no peers, rejecting a known peer jumping from 0 to `MaxUint64`, and the k+1 boundary; logs include `reference_block` and `max_plausible_block`.

<sup>Written for commit c9bf818dfd5146785ef252b7b89be833e6e3753f. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved peer-tip validation using reference-based plausibility checks, reducing false rejections and better handling bootstrap/initial-sync cases; logs now include reference and max-plausible block info.

* **Tests**
  * Expanded tests for peer validation edge cases: initial sync, known-peer jumps, boundary conditions, and security-parameter edge values.

### CODE REVIEW

File: chainselection/selector.go [hunk 1]
```diff
@@ -150,22 +159,44 @@ func (cs *ChainSelector) UpdatePeerTip(
...
+					safeAddUint64(referenceBlock, cs.securityParam)
+			} else if len(cs.peerTips) > 0 {
+				// Case 2: new peer — check against best known
+				for _, pt := range cs.peerTips {
+					if pt.Tip.BlockNumber > referenceBlock {
+						referenceBlock = pt.Tip.BlockNumber
+					}
+				}
+				rejectTip = tip.BlockNumber >
+					safeAddUint64(referenceBlock, cs.securityParam)
```
[the remark this answers is not in the document]
Comment (wolf31o2) [reply chainselection/selector.go]: The finding is not valid. Here's why:                                                                                 
                                                                                                                        
  1. Wrong function identified. Lines 180-188 are in UpdatePeerTip, not selectBestChainLocked. And selectBestChainLocked
   (line 464) already skips stale peers at line 487 — it doesn't compute a referenceBlock at all.                       
                                                                                                                        
  2. The proposed fix would be harmful. The referenceBlock loop in UpdatePeerTip Case 2 (lines 182-185) picks the       
  highest known peer tip to establish a plausibility ceiling for newcomers. If we skip stale peers:

  - All existing peers happen to be stale (common after connectivity blip) → referenceBlock stays 0 → any newcomer with
  tip.BlockNumber > k is rejected → legitimate peers at block 500,000 are rejected because 500,000 > 2,160. The node
  can't recover.
  - The fix would need to also handle the "all peers stale" fallback as a bootstrap case, which adds complexity for no
  gain.

  3. Current behavior is correct. A stale peer's tip was valid within the last 120s (2x StaleTipThreshold). Its block
  number is still a valid reference for what the network looks like. cleanupStalePeers removes truly stale entries (2x
  threshold), and selectBestChainLocked already skips stale peers from chain selection. The plausibility check in
  UpdatePeerTip is deliberately more permissive — it only needs to catch obviously-spoofed tips, not enforce freshness.

  No changes needed.
